stripe/_error.py

```diff
@@ -43,7 +46,9 @@ def __init__(
         self.headers = headers or {}
         self.code = code
         self.request_id = self.headers.get("request-id", None)
-        self.error = self.construct_error_object()
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", DeprecationWarning)
+            self.error = self.construct_error_object()
```

Comment:
nit: create a private version and use here?

Reply:
Ahhh, you mean a private version that call the public version but swallows the error? I didn't think of that, I'll make that change throughout. It will make the follow-up easier.

Reply:
Hm, I meant what you did with empty_list

```
 def empty_list(
        cls,
        api_key: Optional[str] = None,
        stripe_version: Optional[str] = None,
        stripe_account: Optional[str] = None,
    ) -> Self:
        return cls._empty_list(
            api_key=api_key,
            stripe_version=stripe_version,
            stripe_account=stripe_account,
        )

    @classmethod
    def _empty_list(
        cls,
        api_key: Optional[str] = None,
        stripe_version: Optional[str] = None,
        stripe_account: Optional[str] = None,
    ) -> Self:
```

Add _construct_error_object with an implementation, call it from deprecated construct_error_object but call _construct_error_object internally.

Reply:
Ok, I'll make the change. (This is option 1 vs. option 2 in my PR description, by the way) I don't think it makes a whole lot of sense to be worried about subclasses here.

Reply:
Done in [Address review](https://github.com/stripe/stripe-python/pull/1101/commits/cc8e237402eb70696bad6cd98ecb21e2093db4db)

Reply:
Oh, that [caused tests to fail](https://github.com/stripe/stripe-python/actions/runs/7174803441/job/19536938718?pr=1101#step:6:57) because *we* actually do subclass. So I went with "a private version that calls the public version but swallows the error". I think a user defining their own StripeError subclass is fairly unlikely but I feel more comfortable with the more conservative option.